common/libs/network_protocol/network_protocol.proto

```diff
@@ -1,128 +1,202 @@
+
+  // Alarm if tidal volume falls outside this range.
+  required uint32 alarm_lo_tidal_volume_ml = 10;
+  required uint32 alarm_hi_tidal_volume_ml = 11;
+
+  // Alarm if respiratory rate falls outside this range.
+  required uint32 alarm_lo_breaths_per_min = 12;
+  required uint32 alarm_hi_breaths_per_min = 13;
 }
 
-// sent in ControllerData packet in response to get/set commands
-message ControllerValue { optional float value = 1; }
+// See
+// https://docs.google.com/presentation/d/1DA1BQlCj8wNv60pGGzrldPfZBmr6WAH4WbrD4W7i43U
+// for explanation of vent modes.
+enum VentMode {
+  OFF = 0;
+
+  // Every breath is triggered by the machine, at a fixed rate.
+  //
+  // Operational parameters:
+  //
+  //   FiO2 - percent_inspiratory_o2
+  //   PEEP - peep_cm_h2o
+  //   RR   - breaths_per_min
+  //   PIP  - pip_cm_h2o
+  //   I:E  - inspiratory_expiratory_ratio
+  //
+  // Alarm parameters:
+  //
+  //   alarm_lo_tidal_volume_ml
+  //   alarm_hi_tidal_volume_ml
+  //
+  PRESSURE_CONTROL = 1;
+
+  // Patients can trigger a breath for themselves so long as they exceed the
+  // minimum respiration rate.  Ventilator triggers exhalation at a fixed point
+  // after inhalation.
+  //
+  // Operational parameters:
+  //
+  //   FiO2      - percent_inspiratory_o2
+  //   PEEP      - peep_cm_h2o
+  //   min RR    - breaths_per_min (note, interpreted as a minimum here)
+  //   PIP       - pip_cm_h2o
+  //   I:E       - inspiratory_expiratory_ratio
+  //   P-trigger - inspiratory_trigger_cm_h2o
+  //
+  // Alarm parameters:
+  //
+  //   alarm_lo_tidal_volume_ml
+  //   alarm_hi_tidal_volume_ml
+  //   alarm_hi_breaths_per_min
+  //
+  PRESSURE_ASSIST = 2;
+
+  // All inhalations and exhalations are triggered by the patient.
+  //
+  // Operational parameters:
+  //
+  //   FiO2      - percent_inspiratory_o2
+  //   PEEP      - peep_cm_h2o
+  //   PIP       - pip_cm_h2o
+  //   P-trigger - inspiratory_trigger_cm_h2o
+  //   V-trigger - expiratory_trigger_ml_per_min
+  //
+  // Alarm parameters:
+  //
+  //   alarm_lo_tidal_volume_ml
+  //   alarm_hi_tidal_volume_ml
+  //   alarm_lo_breaths_per_min
+  //   alarm_hi_breaths_per_min
+  PRESSURE_SUPPORT = 3;
+}
 
-// All packets sent from controller are wrapped in ControllerData packet
-message ControllerData {
-  required ControllerMsgType msg_type = 1;
-  required CommandType response_to_cmd = 2;
+message SensorReadings {
+  required float pressure_cm_h2o = 1;
+  required float volume_ml = 2;
+  required float flow_ml_per_min = 3;
+}
 
-  oneof payload {
-    ControllerStatus status = 3;
-    ControllerIdentification identification = 4;
-    ControllerValue value = 5;
-    ControlerAlarm alert = 6;
-  }
+enum AlarmType {
+  LO_RESPIRATORY_RATE = 1;
+  HI_RESPIRATORY_RATE = 2;
+  LO_TIDAL_VOLUME = 3;
+  HI_TIDAL_VOLUME = 4;
 }
 
-message Packet {
-  oneof payload {
-    Command cmd = 1;
-    GuiAck gui_ack = 2;
-    ControllerData data = 3;
-  }
+message Alarm {
+  required uint64 start_time = 1;
```

Comment:
Is this millis of the controller's clock or the GUI's?

Reply:
Good point, commented.